File: gencode/java/udmi/schema/Event.java

package udmi.schema;

import javax.annotation.processing.Generated;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Event
 * <p>
 * Container object for all event schemas, not directly used.
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "system",
    "pointset",
    "discovery"
})
@Generated("jsonschema2pojo")
public class Event {

    /**
     * System Event
     * <p>
     * Used for system events such as logging. [System Event Documentation](../docs/messages/system.md#event)
     * 
     */
    @JsonProperty("system")
    @JsonPropertyDescription("Used for system events such as logging. [System Event Documentation](../docs/messages/system.md#event)")
    public SystemEvent system;
    /**
     * Pointset Event
     * <p>
     * A set of points reporting telemetry data. [Pointset Event Documentation](../docs/messages/pointset.md#telemetry)
     * 
     */
    @JsonProperty("pointset")
    @JsonPropertyDescription("A set of points reporting telemetry data. [Pointset Event Documentation](../docs/messages/pointset.md#telemetry)")
    public PointsetEvent pointset;
    /**
     * Discovery Event
     * <p>
     * [Discovery result](../docs/specs/discovery.md) with implicit enumeration
     * 
     */
    @JsonProperty("discovery")
    @JsonPropertyDescription("[Discovery result](../docs/specs/discovery.md) with implicit enumeration")
    public DiscoveryEvent discovery;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.pointset == null)? 0 :this.pointset.hashCode()));
        result = ((result* 31)+((this.system == null)? 0 :this.system.hashCode()));
        result = ((result* 31)+((this.discovery == null)? 0 :this.discovery.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof Event) == false) {
            return false;
        }
        Event rhs = ((Event) other);
        return ((((this.pointset == rhs.pointset)||((this.pointset!= null)&&this.pointset.equals(rhs.pointset)))&&((this.system == rhs.system)||((this.system!= null)&&this.system.equals(rhs.system))))&&((this.discovery == rhs.discovery)||((this.discovery!= null)&&this.discovery.equals(rhs.discovery))));
    }

}

File: gencode/python/udmi/schema/event.py
"""Generated class for event.json"""
from .event_system import SystemEvent
from .event_pointset import PointsetEvent
from .event_discovery import Discovery


class Event:
  """Generated schema class"""

  def __init__(self):
    self.system = None
    self.pointset = None
    self.discovery = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = Event()
    result.system = SystemEvent.from_dict(source.get('system'))
    result.pointset = PointsetEvent.from_dict(source.get('pointset'))
    result.discovery = Discovery.from_dict(source.get('discovery'))
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = Event.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.system:
      result['system'] = self.system.to_dict() # 4
    if self.pointset:
      result['pointset'] = self.pointset.to_dict() # 4
    if self.discovery:
      result['discovery'] = self.discovery.to_dict() # 4
    return result
